Sources/Sandbox.Game/Game/Entities/MyEntity.cs

```diff
@@ -100,6 +100,13 @@ public MyGameLogicComponent GameLogic
             set { Components.Add<MyGameLogicComponent>(value); }
         }
 
+		private HashSet<MyGameLogicComponent> m_gameLogicSet = null;
```

Comment:
Hm, that could also be achieved by introducing a MyCompoundGameLogicComponent that has the collection and would be used as soon as there are more than one MyGameLogicComponents for the same entity.

Or maybe it would be enough to just introduce 3 delegates for the three Update*() methods and attach the set of MyGameLogicComponents to them? That could also serve as a more fine-grained way for update-notifications. If a component isn't interested in Update*100 it isn't registered to that delegate (by some opt-out or opt-in mechanism).

Now that I look at it again, is HashSet appropriate here? How is equality of the elements determined?